### ISSUE TITLE
quantize seems to exit with errorlevel 0 on errors

### ISSUE THREAD

**Issue by maddes8cht** [ggml-org/llama.cpp#3361]
# Expected Behavior

When using `quantize`, for example, there is an error when using K-quants on falcon 7b models with the message :
```
get_k_quant_type : tensor cols 4544 x 4672 are not divisible by 256, required for k-quants
llama_model_quantize: failed to quantize: Unsupported tensor size encountered

main: failed to quantize model from 'ggml-h2oai-h2ogpt-gm-oasst1-multilang-2048-falcon-7b-f32.gguf'.
```
I would expect that the error level on exit from the program is not `0`.
I would like to make further actions dependent on the errorlevel in a batchfile.

# Current Behavior

Errorlevel seems to always be `0` when quantize exits.

# Environment and Context

This behaviour is under Windows 10 in a cmd window.

I took a quick look into qantize.cpp and the included common.cpp and llama.cpp, but there seems to be exceptions are thrown as one would expect.
However, i didn't find a place where the exceptions are being cought, maybe they are not thrown further?

**Comment by BautistaJorge616** [comment]
can i work on this?

**Comment by ggerganov** [comment]
Yes, update `llama_model_quantize_internal()` to propagate a return value

**Comment by paulgekeler** [comment]
I just checked and I am pretty sure this is fixed now. Exceptions are caught in `llama_model_quantize()` which returns 1 upon exception and  so does `main()` of `quantize.cpp`. Or am I missing something?
